### PR TITLE
update vsphere arguments to reflect changes in 0.4.2 provider

### PR DESCRIPTION
**Pull request by jonmosco**
Commit https://github.com/coreos/tectonic-installer/commit/5e662e5c3561586d1ba0fe48ed24b88b2f954079 bumped vSphere provided from 0.2.2 -> 0.4.2.  

Changes have been made to the vphere_folder.  Change datacenter to datacenter_id.  Updated provider.tf and variables.tf to reflect changes.

Default variable to be "vm" since this will primarily be used to create VMs.

### CODE REVIEW

**Comment by jonmosco** [comment]
The changes made with the recent patch for datacenters is still not working with 0.4.2 due to changes made to the vsphere_folder:

From the terraform providers changelog regarding 0.4.0:

BREAKING CHANGES:

The vsphere_folder resource has been re-written, and its configuration is significantly different. See the resource documentation for more details. Existing state will be migrated. (#179)

I am getting errors on new builds that require 0.4.2:

2 error(s) occurred:

* vsphere_folder.tectonic_vsphere_folder: "type": required field is not set
* vsphere_folder.tectonic_vsphere_folder: : invalid or unknown key: datacenter

File: platforms/vmware/provider.tf [hunk 1]
```diff
@@ -5,6 +5,7 @@ provider "vsphere" {
 }
 
 resource "vsphere_folder" "tectonic_vsphere_folder" {
-  path       = "${var.tectonic_vmware_folder}"
-  datacenter = "${var.tectonic_vmware_worker_datacenters[0]}"
-}
+  path          = "${var.tectonic_vmware_folder}"
+  type          = "${var.tectonic_vmware_type}"
+  datacenter_id = "${var.tectonic_vmware_datacenter}"
+}
```
Comment (squat) [inline platforms/vmware/provider.tf]: Please ensure there is an appropriate newline character at the end of this file. You may need to tweak your editor's config for this.

File: platforms/vmware/variables.tf [hunk 2]
```diff
@@ -25,6 +25,28 @@ variable "tectonic_vmware_folder" {
   description = "vSphere Folder to create and add the Tectonic nodes"
 }
 
+variable "tectonic_vmware_type" {
+  type        = "string"
+  description = "The type of folder to create. Allowed options: datacenter,host,vm, datastore, and network."
```
Comment (squat) [inline platforms/vmware/variables.tf]: Please add a single space character after the commas between items in the list.

**Comment by jonmosco** [comment]
Closed.  Opened https://github.com/coreos/tectonic-installer/pull/2285

**Pull request closed without merging**